pkg/cli/admin/mustgather/mustgather_test.go

```diff
@@ -478,3 +478,41 @@ sync && echo 'Caches written to disk'`,
 		})
 	}
 }
+
+func TestNewPod(t *testing.T) {
+	tests := []struct {
+		name        string
+		node        string
+		affinity    *corev1.Affinity
+		affinitySet bool
+	}{
+		{
+			name:        "node set, affinity provided, affinity not set",
+			node:        "node1",
+			affinity:    buildNodeAffinity([]string{"node2"}),
+			affinitySet: false,
+		},
+		{
+			name:        "node not set, affinity provided, affinity set",
+			node:        "",
+			affinity:    buildNodeAffinity([]string{"node2"}),
+			affinitySet: true,
```

Comment:
I would find it nicer and more generic to use `expectedPod` and compare using `cmp.Equal/cmp.Diff`. Otherwise this test just tests the affinity. Which is also ok, but in that case I would rename the test.